File: Source/WebDemo/SerenePSM/SerenePSM.Web/Modules/ServerTypes/Default/Pfk7155Columns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Pfk7155Row } from "./Pfk7155Row";

export interface Pfk7155Columns {
    MachineCode: Column<Pfk7155Row>;
    DevelopmentCode: Column<Pfk7155Row>;
    AccCode: Column<Pfk7155Row>;
    TagNo: Column<Pfk7155Row>;
    Model: Column<Pfk7155Row>;
    SeMachineType: Column<Pfk7155Row>;
    CdMachineType: Column<Pfk7155Row>;
    MachineName: Column<Pfk7155Row>;
    MachineNameSimply: Column<Pfk7155Row>;
    SeFactory: Column<Pfk7155Row>;
    CdFactory: Column<Pfk7155Row>;
    SeSubProcess: Column<Pfk7155Row>;
    CdSubProcess: Column<Pfk7155Row>;
    MachineRpm: Column<Pfk7155Row>;
    MachineCapacity: Column<Pfk7155Row>;
    RotationDayProduction: Column<Pfk7155Row>;
    RotationOnceProduction: Column<Pfk7155Row>;
    RotationOnceTimeProduction: Column<Pfk7155Row>;
    CheckApplyLiquid: Column<Pfk7155Row>;
    RateLiquid: Column<Pfk7155Row>;
    Wdcpmdtc: Column<Pfk7155Row>;
    RateSoda1: Column<Pfk7155Row>;
    RateSoda2: Column<Pfk7155Row>;
    RateSoda3: Column<Pfk7155Row>;
    RateGlauberSalt1: Column<Pfk7155Row>;
    RateGlauberSalt2: Column<Pfk7155Row>;
    RateGlauberSalt3: Column<Pfk7155Row>;
    GroupMachine: Column<Pfk7155Row>;
    HorizontalPosition: Column<Pfk7155Row>;
    VerticalPosition: Column<Pfk7155Row>;
    ChecK1: Column<Pfk7155Row>;
    Check2: Column<Pfk7155Row>;
    CheckFinalProcess: Column<Pfk7155Row>;
    CheckProcess: Column<Pfk7155Row>;
    CheckUse: Column<Pfk7155Row>;
    DisplaySeq: Column<Pfk7155Row>;
    Remark: Column<Pfk7155Row>;
    SeSite: Column<Pfk7155Row>;
    CdSite: Column<Pfk7155Row>;
}

export class Pfk7155Columns extends ColumnsBase<Pfk7155Row> {
    static readonly columnsKey = 'Default.Pfk7155';
    static readonly Fields = fieldsProxy<Pfk7155Columns>();
}

File: Source/WebDemo/SerenePSM/SerenePSM.Web/Modules/Default/Pfk7155/Pfk7155Columns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SerenePSM.Default.Columns;

[ColumnsScript("Default.Pfk7155")]
[BasedOnRow(typeof(Pfk7155Row), CheckNames = true)]
public class Pfk7155Columns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public string MachineCode { get; set; }
    public string DevelopmentCode { get; set; }
    public string AccCode { get; set; }
    public string TagNo { get; set; }
    public string Model { get; set; }
    public string SeMachineType { get; set; }
    public string CdMachineType { get; set; }
    public string MachineName { get; set; }
    public string MachineNameSimply { get; set; }
    public string SeFactory { get; set; }
    public string CdFactory { get; set; }
    public string SeSubProcess { get; set; }
    public string CdSubProcess { get; set; }
    public decimal MachineRpm { get; set; }
    public decimal MachineCapacity { get; set; }
    public decimal RotationDayProduction { get; set; }
    public decimal RotationOnceProduction { get; set; }
    public decimal RotationOnceTimeProduction { get; set; }
    public string CheckApplyLiquid { get; set; }
    public decimal RateLiquid { get; set; }
    public string Wdcpmdtc { get; set; }
    public decimal RateSoda1 { get; set; }
    public decimal RateSoda2 { get; set; }
    public decimal RateSoda3 { get; set; }
    public decimal RateGlauberSalt1 { get; set; }
    public decimal RateGlauberSalt2 { get; set; }
    public decimal RateGlauberSalt3 { get; set; }
    public decimal GroupMachine { get; set; }
    public decimal HorizontalPosition { get; set; }
    public decimal VerticalPosition { get; set; }
    public string ChecK1 { get; set; }
    public string Check2 { get; set; }
    public string CheckFinalProcess { get; set; }
    public string CheckProcess { get; set; }
    public string CheckUse { get; set; }
    public decimal DisplaySeq { get; set; }
    public string Remark { get; set; }
    public string SeSite { get; set; }
    public string CdSite { get; set; }
}